target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-mr8300.dts

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+/dts-v1/;
+
+#include "qcom-ipq4019-xx8300.dtsi"
+
+/ {
+	model = "Linksys MR8300 (Dallas)";
+	compatible = "linksys,mr8300", "qcom,ipq4019";
+
+
+	aliases {
+		led-boot = &led_blue;
+		led-failsafe = &led_red;
+		led-running = &led_blue;
+		led-upgrade = &led_amber;
+		serial0 = &blsp1_uart1;
+	};
+
+		// Top panel LEDs, above Linksys logo
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_red: red {
+			label = "mr8300:red";
+			gpios = <&tlmm 47 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_amber: amber {
+			label = "mr8300:amber";
+			gpios = <&tlmm 22 GPIO_ACTIVE_HIGH>;
+			panic-indicator;
+		};
+
+		led_blue: blue {
+			label = "mr8300:blue";
+			gpios = <&tlmm 46 GPIO_ACTIVE_HIGH>;
+		};
+
+		// On back panel, above USB socket
+
+		led_usb: usb {
+			label = "mr8300:green:usb";
+			gpios = <&tlmm 61 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&usb3_port1>, <&usb3_port2>,
+					  <&usb2_port1>;
+			linux,default-trigger = "usbport";
+		};
+	};
+
+    keys {
+		compatible = "gpio-keys";
+
+		button@0 {
```

Comment:
why the unit-address here and below?

Reply:
This was an exact extract from the ea8300 DTS file, I have been trying to keep them as close as possible. I assume when Jeff originally built the DTS file he saw a need to have this there?

Reply:
well, the unit-address and duplicated node-name unfortunately flew under the radar during the EA8300 review. Oh well, I guess this is something for the next big refresh for when the next LTS kernel ( 5.10, 5.11, 5.12, ...?) hits. Let's hope I don't forget it until then.

Reply:
... I don't think this has been fixed? The EA8300 is wrong too, but as I said that "flew by" and should be looked after later.

Reply:
@chunkeey what are you seeing here that's duplicated? The definition of "compatible = "gpio-keys";" or button@0 vs @1?

Reply:
@chunkeey , without the below two entries, the wps and reset buttons do not register being pushed, they are unique @0 and @1:

```
    keys {
		compatible = "gpio-keys";

		button@0 {
			label = "reset";
			linux,code = <KEY_RESTART>;
			gpios = <&tlmm 50 GPIO_ACTIVE_LOW>;
		};

		button@1 {
			label = "wps";
			linux,code = <KEY_WPS_BUTTON>;
			gpios = <&tlmm 18 GPIO_ACTIVE_LOW>;
		};
	};
```

Reply:
yes exactly :)  that's why node-name is "duplicated". This causes the latter entry to overwrite the properties in the former. you need unique node-names and drop the wrong unit-addresses and you are golden. 

Maybe this helps, https://elinux.org/Device_Tree_Usage#Node_Names to understand what I'm on about?!

Reply:
> yes exactly :) that's why node-name is "duplicated". This causes the latter entry to overwrite the properties in the former. you need unique node-names and drop the wrong unit-addresses and you are golden.
> 
> Maybe this helps, https://elinux.org/Device_Tree_Usage#Node_Names to understand what I'm on about?!

I promise I am not trying to be difficult, but I am failing to see what entry is "duplicated"? One entry is button@**0** specific for the reset button and the other entry is button@**1** specific to the wps button. I am really failing to see it :-/

Would you be able to draft up roughly what an example of what you mean would look like, and past it as a copy snippet so I can visually see what you are referring to ?

Reply:
Oh, wait a minute........ :-) I think I see what you mean. I am looking at another routers DTS file as an example (the R7800 since I am sure that is correct). I will work on this now...

Reply:
Okay, this should be sorted out now.

Reply:
@chunkeey Do you want me to include these updates for the EA8300 as well in this PR? Let me know....

Reply:
@chunkeey I went ahead and cleaned this up for the EA8300 as well ;-)